weblate/machinery/google.py

```diff
@@ -99,3 +101,46 @@ def get_error_message(self, exc):
                 pass
 
         return super().get_error_message(exc)
+
+
+class GoogleTranslationAdvanced(MachineTranslation):
+    """Google Translate API v3 machine translation support."""
+
+    name = "Google Translate Advanced"
+    max_score = 90
+
+    def __init__(self):
+        """Check configuration."""
+        super().__init__()
+        if settings.MT_GOOGLE_CREDENTIALS is None:
+            raise MissingConfiguration("Google Translate requires API key")
+        if settings.MT_GOOGLE_PROJECT is None:
+            raise MissingConfiguration(
+                "You have to specify Google Cloud project "
+                "affiliated with provided credentials"
+            )
+        credentials = service_account.Credentials.from_service_account_file(
+            settings.MT_GOOGLE_CREDENTIALS
+        )
+        self.client = TranslationServiceClient(credentials=credentials)
+        self.parent = self.client.location_path(settings.MT_GOOGLE_PROJECT, "global")
```

Comment:
Shouldn't the second part be configurable as well? (I have no clue, just asking)

Reply:
Yes, it is possible to have a location in project. However, given that global is default, it works for anyone having empty translation project. In case of having the `parent` configurable, i'd suggest the whole location string than having. That again seems more complicated.
Or i can see option to add either project name or full path. We can determine which one it is by perhaps using a regular expression. This one is up to you.

Reply:
I did some digesting, and i came to result. It seems from this error `InvalidArgument: 400 Invalid 'parent'.; Invalid location name. Unsupported location 'local'. Must be 'us-central1' or 'global'.
`
So the location can either be `'global'` or `'us-central1'`
We can assume that the location depends on cloud owner google user account. However ia assume we should try catch the correct location and save it. But i have no idea where.

Reply:
Now, i do not know if only my account can either use `us-central` and `global`, but i can use both.

Reply:
Let's make it configurable with "global" being default. I did similar change for Microsoft Translate in 037e50719718878217b74f9dc3b03be008db168a.